tests/test_vqe.py

```diff
@@ -749,18 +747,14 @@ def test_error_non_expval_measurement(self):
         def circuit():
             return qml.sample(H)
 
-        with pytest.raises(
-            ValueError, match="Only a single expectation of a Hamiltonian observable "
-        ):
+        with pytest.raises(ValueError, match="expectation of a Hamiltonian observable"):
```

Comment:
Wait, the codecov complaints are there because I introduce a number of untested `NotImplemented` errors, no? This one should be fine... 

About the `NotImplemented` ones, I could make some exotic tests to cover them, but I am inclined to just overwrite and merge, since we should not just add useless tests.

Reply:
Ohh I see, is it not catching them because of the chaining perhaps? I ran into this somewhere else, where an error produced an error and I was trying to test for the original error, but pytest just wasn't having it. In any case yeah fine to leave as-is, then.

Reply:
No, I think I am literally not having test cases for this. I could cook up something with monkeypatching, but it would be very artificial...